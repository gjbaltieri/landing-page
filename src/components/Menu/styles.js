import Styled, { css } from 'styled-components';
import { Container as SectionContainer } from '../SectionContainer/styles';
import { Title as Heading } from '../Header/Title';

const MenuClose = (theme) => css`
  visibility: visible;
  opacity: 1;
`;

export const Container = Styled.div`
  ${({ theme, visible }) => css`
    position: fixed;
    heigth: 10px;
    z-index: 5;
    top: 0;
    bottom: 0;
    left: 0;
    right: 0;
    transition: all 300ms ease-in-out;

    > ${SectionContainer} {
      padding-top: 0;
      padding-bottom: 0;
      background: ${theme.color.background.black};
      border-bottom: solid ${theme.color.background.black};
    }
    & ${Heading} {
      margin-top: 0;
      margin-bottom: 0;
    }

    @media ${theme.media.maxWidth768px} {
      height: 100vh;
      visibility: hidden;
      opacity: 0;
      ${visible && MenuClose(theme)};

      > ${SectionContainer} {
        display: grid;
        overflow-y: auto;
        grid-template-columns: 1fr;
        grid-template-rows: 1fr;
        height: 100vh;
        align-items: center;
        background: ${theme.color.background.white};
      }

      & ${Heading} {
        display: flex;
        justify-content: center;
        padding-bottom: ${theme.padding.large};
      }
    }
  `}`;

export const MenuContainer = Styled.div`
${({ theme }) => css`
  display: flex;
  justify-content: space-between;
  align-items: center;

  @media ${theme.media.maxWidth768px} {
    display: block;
    text-align: center;
  } ;
`}`;

export const MenuIcon = Styled.div`
${({ theme, visible }) => css`
  display: none;
  border: none;

  @media ${theme.media.maxWidth768px} {
    display: flex;
    position: fixed;
    top: 2rem;
    right: 2rem;
    z-index: 6;
    width: 3.5rem;
    heigth: 3.5rem;
    background: none;
    pointer-events: ${visible ? 'none' : 'all'};
  }
`}`;
