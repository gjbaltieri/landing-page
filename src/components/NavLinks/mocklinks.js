export const links = [
  {
    children: 'google',
    link: 'http://google.com',
    newTab: false,
  },
  {
    children: 'twitter',
    link: 'http://twitter.com',
    newTab: false,
  },
  {
    children: 'uol',
    link: 'http://uol.com',
    newTab: false,
  },
];
