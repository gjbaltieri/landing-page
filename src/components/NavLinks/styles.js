import Styled, { css } from 'styled-components';
export const Container = Styled.nav`
${({ theme }) => css`
  display: flex;
  flex-flow: row wrap;
  @media ${theme.media.maxWidth768px} {
    flex-flow: column wrap;
    align-items: center;
  } ;
`}`;
