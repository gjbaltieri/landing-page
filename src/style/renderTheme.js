import { render } from '@testing-library/react';
import { GlobalStyles } from './globalstyle';
import { theme } from './theme';
import { ThemeProvider } from 'styled-components';

export default function renderTheme(children) {
  return render(
    <ThemeProvider theme={theme}>
      {children}
      <GlobalStyles />
    </ThemeProvider>
  );
}
