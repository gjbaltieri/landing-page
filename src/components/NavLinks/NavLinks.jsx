import P from 'prop-types';
import * as Styled from './styles';
import MenuLink from '../MenuLink/MenuLink.jsx';

export default function NavLinks({ links = [] }) {
  return (
    <Styled.Container>
      {links.map((link) => (
        <MenuLink key={link.link} {...link} />
      ))}
    </Styled.Container>
  );
}

NavLinks.prototype = {
  links: P.arrayOf(
    P.shape({
      children: P.string.isRequired,
      link: P.string.isRequired,
      newTab: P.bool,
    })
  ),
};
