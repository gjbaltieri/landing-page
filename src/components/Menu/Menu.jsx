/* eslint-disable react/forbid-foreign-prop-types */
import P from 'prop-types';
import * as Styled from './styles';
import SectionContainer from '../SectionContainer/SectionContainer';
import LogoLink from '../LogoLink/LogoLink';
import NavLinks from '../NavLinks/NavLinks';
import { Menu as MenuIcon } from '@styled-icons/material-outlined/Menu';
import { Close as MenuClose } from '@styled-icons/material-outlined/Close';

import { useState } from 'react';

export default function Menu({ links = [], logoData }) {
  const [menuOpen, setMenuOpen] = useState(false);
  return (
    <>
      <Styled.MenuIcon
        onClick={() => setMenuOpen(!menuOpen)}
        visible={menuOpen}
        aria-label='Open/Close menu'
      >
        {menuOpen ? (
          <MenuClose aria-label='Close menu' />
        ) : (
          <MenuIcon aria-label='Open menu' />
        )}
      </Styled.MenuIcon>
      <Styled.Container visible={menuOpen} onClick={() => setMenuOpen(false)}>
        <SectionContainer>
          <Styled.MenuContainer>
            <LogoLink {...logoData} />
            <NavLinks links={links} />
          </Styled.MenuContainer>
        </SectionContainer>
      </Styled.Container>
    </>
  );
}

Menu.propTypes = {
  ...NavLinks.propTypes,
  logoData: P.shape(LogoLink.propTypes).isRequired,
};
