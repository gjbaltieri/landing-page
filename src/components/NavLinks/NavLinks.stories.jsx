import React from 'react';
import NavLinks from './NavLinks';
import { links } from './mocklinks';

export default {
  title: 'NavLinks',
  component: NavLinks,
  args: { links: links },
  argTypes: {
    links: { type: '' },
  },
};

export const Template = (args) => (
  <div>
    <NavLinks {...args} />
  </div>
);
