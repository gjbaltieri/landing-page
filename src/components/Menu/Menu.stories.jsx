import React from 'react';
import Menu from './Menu';
import { links } from '../NavLinks/mocklinks.js';

export default {
  title: 'Menu',
  component: Menu,
  args: {
    links: links,
    logoData: { text: 'logo', link: '#target', srcImg: '' },
  },
  argTypes: {
    children: { type: 'string' },
  },
};

export const Template = (args) => {
  return (
    <div>
      <Menu {...args} />
    </div>
  );
};
