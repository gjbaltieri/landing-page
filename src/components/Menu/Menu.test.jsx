import 'jest-styled-components';
import '@testing-library/jest-dom';
import { fireEvent, screen } from '@testing-library/react';
import renderTheme from '../../style/renderTheme';
import Menu from './Menu';
import { theme } from '../../style/theme';
import { links } from '../NavLinks/mocklinks';

const logoData = {
  text: 'Logo',
  link: '#Target',
};

describe('<Menu/>', () => {
  test('should render Menu with default values', () => {
    const { container } = renderTheme(
      <Menu logoData={logoData} links={links} />
    );
    expect(screen.getByRole('heading', { name: 'Logo' })).toBeInTheDocument();
    expect(
      screen.getByRole('navigation', { name: 'Main menu' })
    ).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  test('should render Menu mobile with button', () => {
    const { container } = renderTheme(<Menu logoData={logoData} />);
    const button = screen.getByLabelText('Open/Close menu');
    // eslint-disable-next-line testing-library/no-node-access
    const menuContainer = button.nextSibling;
    expect(button).toHaveStyleRule('display', 'none');
    expect(button).toHaveStyleRule('display', 'flex', {
      media: theme.media.maxWidth768px,
    });
    expect(menuContainer).toHaveStyleRule('opacity', '0', {
      media: theme.media.maxWidth768px,
    });
    expect(screen.getByLabelText('Open menu')).toBeInTheDocument();

    fireEvent.click(button);
    expect(menuContainer).toHaveStyleRule('opacity', '1', {
      media: theme.media.maxWidth768px,
    });
    expect(screen.getByLabelText('Close menu')).toBeInTheDocument();

    fireEvent.click(menuContainer);
    expect(menuContainer).toHaveStyleRule('opacity', '0', {
      media: theme.media.maxWidth768px,
    });
    expect(screen.getByLabelText('Open menu')).toBeInTheDocument();
  });

  it('should <Menu/> not render links', () => {
    const { container } = renderTheme(<Menu logoData={logoData} />);
    expect(
      // eslint-disable-next-line testing-library/no-node-access
      screen.queryByRole('navigation', { name: 'Main menu' }).firstChild
    ).not.toBeInTheDocument();

    expect(container).toMatchSnapshot();
  });
});
