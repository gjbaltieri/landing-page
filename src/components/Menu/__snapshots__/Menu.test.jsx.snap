// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Menu/> should <Menu/> not render links 1`] = `
.c4 {
  max-width: 120rem;
  margin: 0 auto;
  padding: 2.4rem;
}

.c7 {
  color: #333333;
  font-size: 1.6rem;
  font-weight: 900;
  text-transform: uppercase;
}

.c2 {
  position: fixed;
  heigth: 10px;
  z-index: 5;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

.c2 > .c3 {
  padding-top: 0;
  padding-bottom: 0;
  background: #0D0D0D;
  border-bottom: solid #0D0D0D;
}

.c2 .c6 {
  margin-top: 0;
  margin-bottom: 0;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 {
  display: none;
  border: none;
}

.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 1.6rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
}

.c8 > img {
  max-height: 2rem;
}

.c9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: row wrap;
  -ms-flex-flow: row wrap;
  flex-flow: row wrap;
}

.c1 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

@media (max-width:768px) {
  .c7 {
    font-size: 4.0rem;
  }
}

@media (max-width:768px) {
  .c2 {
    height: 100vh;
    visibility: hidden;
    opacity: 0;
  }

  .c2 > .c3 {
    display: grid;
    overflow-y: auto;
    grid-template-columns: 1fr;
    grid-template-rows: 1fr;
    height: 100vh;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
    background: #F2F2F2;
  }

  .c2 .c6 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
    padding-bottom: 3.2rem;
  }
}

@media (max-width:768px) {
  .c5 {
    display: block;
    text-align: center;
  }
}

@media (max-width:768px) {
  .c0 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    position: fixed;
    top: 2rem;
    right: 2rem;
    z-index: 6;
    width: 3.5rem;
    heigth: 3.5rem;
    background: none;
    pointer-events: all;
  }
}

@media (max-width:768px) {
  .c9 {
    -webkit-flex-flow: column wrap;
    -ms-flex-flow: column wrap;
    flex-flow: column wrap;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
  }
}

<div>
  <div
    aria-label="Open/Close menu"
    class="c0"
  >
    <svg
      aria-hidden="true"
      aria-label="Open menu"
      class="c1"
      fill="currentColor"
      focusable="false"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
      />
    </svg>
  </div>
  <div
    class="c2"
  >
    <div
      class="c3 c4"
    >
      <div
        class="c5"
      >
        <h1
          class="c6 c7"
          font-size="base"
        >
          <a
            class="c8"
            href="#Target"
          >
            Logo
          </a>
        </h1>
        <nav
          aria-label="Main menu"
          class="c9"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`<Menu/> should render Menu with default values 1`] = `
.c4 {
  max-width: 120rem;
  margin: 0 auto;
  padding: 2.4rem;
}

.c7 {
  color: #333333;
  font-size: 1.6rem;
  font-weight: 900;
  text-transform: uppercase;
}

.c2 {
  position: fixed;
  heigth: 10px;
  z-index: 5;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

.c2 > .c3 {
  padding-top: 0;
  padding-bottom: 0;
  background: #0D0D0D;
  border-bottom: solid #0D0D0D;
}

.c2 .c6 {
  margin-top: 0;
  margin-bottom: 0;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 {
  display: none;
  border: none;
}

.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 1.6rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
}

.c8 > img {
  max-height: 2rem;
}

.c9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: row wrap;
  -ms-flex-flow: row wrap;
  flex-flow: row wrap;
}

.c10 {
  font-size: 1.6rem;
  padding: 1.6rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #333333;
  position: relative;
}

.c10:before {
  content: '';
  background: #54b3d6;
  display: block;
  position: absolute;
  bottom: -3px;
  left: 0;
  width: 0;
  height: 3px;
}

.c10::after {
  content: '';
  position: absolute;
  bottom: 0.75rem;
  left: 0;
  width: 100%;
  height: 2px;
  background-color: #0D0D0D;
  -webkit-transform: scaleX(0);
  -ms-transform: scaleX(0);
  transform: scaleX(0);
  -webkit-transform-origin: center;
  -ms-transform-origin: center;
  transform-origin: center;
  -webkit-transition: -webkit-transform 0.3s ease;
  -webkit-transition: transform 0.3s ease;
  transition: transform 0.3s ease;
}

.c10:hover::after {
  -webkit-transform: scaleX(1);
  -ms-transform: scaleX(1);
  transform: scaleX(1);
}

.c1 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

@media (max-width:768px) {
  .c7 {
    font-size: 4.0rem;
  }
}

@media (max-width:768px) {
  .c2 {
    height: 100vh;
    visibility: hidden;
    opacity: 0;
  }

  .c2 > .c3 {
    display: grid;
    overflow-y: auto;
    grid-template-columns: 1fr;
    grid-template-rows: 1fr;
    height: 100vh;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
    background: #F2F2F2;
  }

  .c2 .c6 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
    padding-bottom: 3.2rem;
  }
}

@media (max-width:768px) {
  .c5 {
    display: block;
    text-align: center;
  }
}

@media (max-width:768px) {
  .c0 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    position: fixed;
    top: 2rem;
    right: 2rem;
    z-index: 6;
    width: 3.5rem;
    heigth: 3.5rem;
    background: none;
    pointer-events: all;
  }
}

@media (max-width:768px) {
  .c9 {
    -webkit-flex-flow: column wrap;
    -ms-flex-flow: column wrap;
    flex-flow: column wrap;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
  }
}

<div>
  <div
    aria-label="Open/Close menu"
    class="c0"
  >
    <svg
      aria-hidden="true"
      aria-label="Open menu"
      class="c1"
      fill="currentColor"
      focusable="false"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
      />
    </svg>
  </div>
  <div
    class="c2"
  >
    <div
      class="c3 c4"
    >
      <div
        class="c5"
      >
        <h1
          class="c6 c7"
          font-size="base"
        >
          <a
            class="c8"
            href="#Target"
          >
            Logo
          </a>
        </h1>
        <nav
          aria-label="Main menu"
          class="c9"
        >
          <a
            class="c10"
            href="http://google.com"
            target="_self"
          >
            google
          </a>
          <a
            class="c10"
            href="http://twitter.com"
            target="_self"
          >
            twitter
          </a>
          <a
            class="c10"
            href="http://uol.com"
            target="_self"
          >
            uol
          </a>
        </nav>
      </div>
    </div>
  </div>
</div>
`;
